microovn/ovn/join.go

```diff
@@ -142,13 +142,58 @@ func Join(s *state.State, initConfig map[string]string) error {
 		return fmt.Errorf("Failed to get OVN SB connect string: %w", err)
 	}
 
+	// Encapsulation IP configuration
+
+	// A custom encapsulation IP address can be defined for a join member by a bootstrapping member.
+	// In this case, the custom IP address will be stored in the database by the bootstrapping member
+	// after its creation. This block is typically executed by a `microovn cluster join` command
+	var ovnEncapIp string
+	err = s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
+		item, err := database.GetConfigItem(ctx, tx, fmt.Sprintf("%s.ovn-encap-ip", s.Name()))
+		if err != nil {
+			if err.Error() == "ConfigItem not found" {
+				return nil
+			}
+
+			return err
+		}
+
+		ovnEncapIp = item.Value
+
+		// Remove the config item as we only need it one time for the OVS configuration.
```

Comment:
I'm bit wary of removing the key from the database right away. Are there any benefits in removing it? If there are, I'd suggest removing it as a last step in the `Join` function, otherwise if the `Join` fails after this transaction block, it can not be retried.

Reply:
Do I understand correctly that you will remove the `DeleteConfigItem` then? 'cause I got re-request for review, but I don't see any changes :D

Reply:
Ah no this was just a follow up statement.. sorry for that. But yes, I'll remove the delete operation

Reply:
fixed